File: paper/src/main/java/com/al3x/housing2/Data/StatData.java
package com.al3x.housing2.Data;

import com.al3x.housing2.Instances.Stat;
import lombok.Getter;
import lombok.Setter;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.UUID;
import java.util.stream.Collectors;

@Setter
@Getter
public class StatData {
    private String name;
    private String value; // Can be a string, double, or int

    public StatData() {}

    public StatData(String name, String value) {
        this.name = name;
        this.value = value;
    }

    public String ofString() {
        return value;
    }

    public double ofDouble() {
        return Double.parseDouble(value);
    }

    public int ofInt() {
        return Integer.parseInt(value);
    }

    public static Map<String, List<StatData>> fromHashMap(Map<UUID, List<Stat>> statMap) {
        return statMap.entrySet().stream().collect(Collectors.toMap(
            entry -> entry.getKey().toString(),
            entry -> entry.getValue().stream()
                .filter(stat -> stat.getValue() != null)
                .map(stat -> new StatData(stat.getStatName(), stat.getValue()))
                .collect(Collectors.toList())
        ));
    }

    public static Map<UUID, List<Stat>> toHashMap(Map<String, List<StatData>> statMap) {
        return statMap.entrySet().stream().collect(Collectors.toMap(
            entry -> UUID.fromString(entry.getKey()),
            entry -> entry.getValue().stream()
                .map(statData -> new Stat(statData.getName(), statData.getValue()))
                .collect(Collectors.toList())
        ));
    }

    public static List<StatData> fromList(List<Stat> statList) {
        return statList.stream()
            .filter(stat -> stat.getValue() != null)
            .map(stat -> new StatData(stat.getStatName(), stat.getValue()))
            .collect(Collectors.toList());
    }

    public static List<Stat> toList(List<StatData> statDataList) {
        return statDataList.stream()
            .map(statData -> new Stat(statData.getName(), statData.getValue()))
            .collect(Collectors.toList());
    }
}

File: src/main/java/com/al3x/housing2/Instances/HousingData/StatData.kt
package com.al3x.housing2.Instances.HousingData

import com.al3x.housing2.Instances.Stat
import java.util.*
import kotlin.collections.HashMap

data class StatData(
    val name: String,
    val value: String //Can be a string, double, or int
) {
    //Current 3 types of values that can be stored in a StatData object
    fun ofString(): String {
        return value
    }

    fun ofDouble(): Double {
        return value.toDouble()
    }

    fun ofInt(): Int {
        return value.toInt()
    }

    companion object {
        fun fromHashMap(statMap: HashMap<UUID, List<Stat>>): HashMap<String, List<StatData>> {
            val map = hashMapOf<String, List<StatData>>()
            statMap.forEach { (uuid, statList) ->
                val list = mutableListOf<StatData>()
                statList.forEach {
                    if (it.value == null) return@forEach
                    list.add(StatData(it.statName, it.value))
                }
                map[uuid.toString()] = list
            }
            return map
        }

        fun toHashMap(statMap: HashMap<String, List<StatData>>): HashMap<UUID, List<Stat>> {
            val map = hashMapOf<UUID, List<Stat>>()
            statMap.forEach { (uuid, statList) ->
                val list = mutableListOf<Stat>()
                statList.forEach {
                    list.add(Stat(it.name, it.value))
                }
                map[UUID.fromString(uuid)] = list
            }
            return map
        }

        fun fromList(statList: List<Stat>): List<StatData> {
            val list = mutableListOf<StatData>()
            statList.forEach {
                if (it.value == null) return@forEach
                list.add(StatData(it.statName, it.value))
            }
            return list
        }

        fun toList(statList: List<StatData>): List<Stat> {
            val list = mutableListOf<Stat>()
            statList.forEach {
                list.add(Stat(it.name, it.value))
            }
            return list
        }
    }
}